src/map/pc.cpp

```diff
@@ -2951,7 +2951,7 @@ uint16 pc_getpercentweight(map_session_data& sd, uint32 weight)
 {
 	if (weight == 0)
 		weight = sd.weight;
-	return static_cast<uint16>(weight * 100 / sd.max_weight);
+	return static_cast<uint16>(weight * 100 / std::max(sd.max_weight, static_cast<uint32>(1)));
```

Comment:
You could put the variable type you want to base the max calculation on behind std::max I think.
`std::max<uint32>(sd.max_weight, 1);`
But didn't test.